addons/website_slides/controllers/main.py

```diff
@@ -78,9 +78,14 @@ def _set_completed_slide(self, slide):
         # quiz use their specific mechanism to be marked as done
         if slide.slide_category == 'quiz' or slide.question_ids:
             raise UserError(_("Slide with questions must be marked as done when submitting all good answers "))
-        if slide.website_published and slide.channel_id.is_member:
-            slide.action_set_completed()
-        return True
+        if not slide.can_be_skipped:
+            raise werkzeug.exceptions.Forbidden(_("This slide can not be marked as completed."))
+        slide.action_set_completed()
+
+    def _set_uncompleted_slide(self, slide):
```

Comment:
If we rename the fields (see other comment), we should probably rename those as well (_slide_mark_completed / _slide_mark_uncompleted), same logic for the route methods.